modin/engines/base/io/text/excel_reader.py

```diff
@@ -0,0 +1,202 @@
+            reader.bind_cells()
+            data = PandasExcelParser.get_sheet_data(
+                ws, kwargs.pop("convert_float", True)
+            )
+            # Extract column names from parsed data.
+            column_names = pandas.Index(data[0])
+            index_col = kwargs.get("index_col", None)
+            # Remove column names that are specified as `index_col`
+            if index_col is not None:
+                column_names = column_names.drop(column_names[index_col])
+            # Compute partition metadata upfront so it is uniform for all partitions
+            chunk_size = max(1, (total_bytes - f.tell()) // num_partitions)
+            num_splits = min(len(column_names), num_partitions)
+            kwargs["fname"] = io
+            # Skiprows will be used to inform a partition how many rows come before it.
+            kwargs["skiprows"] = 0
+            row_count = 0
+            data_ids = []
+            index_ids = []
+            dtypes_ids = []
+
+            # Compute column metadata
+            column_chunksize = compute_chunksize(
+                pandas.DataFrame(columns=column_names), num_splits, axis=1
+            )
+            if column_chunksize > len(column_names):
+                column_widths = [len(column_names)]
+                # This prevents us from unnecessarily serializing a bunch of empty
+                # objects.
+                num_splits = 1
+            else:
+                column_widths = [
+                    column_chunksize
+                    if len(column_names) > (column_chunksize * (i + 1))
+                    else 0
+                    if len(column_names) < (column_chunksize * i)
+                    else len(column_names) - (column_chunksize * i)
+                    for i in range(num_splits)
+                ]
+            kwargs["num_splits"] = num_splits
+
+            while f.tell() < total_bytes:
+                print(f.tell())
+                print(total_bytes)
+                args = kwargs
+                args["skiprows"] = row_count + args["skiprows"]
+                args["start"] = f.tell()
+                chunk = f.read(chunk_size)
+                # This edge case can happen when we have reached the end of the data
+                # but not the end of the file.
+                if b"<row" not in chunk:
+                    break
+                row_close_tag = b"</row>"
+                row_count = re.subn(row_close_tag, b"", chunk)[1]
+                break_condition = b"</sheetData>" in chunk
+
+                # Make sure we are reading at least one row.
+                while row_count == 0:
+                    chunk += f.read(chunk_size)
+                    row_count += re.subn(row_close_tag, b"", chunk)[1]
+
+                last_index = chunk.rindex(row_close_tag)
+                f.seek(-(len(chunk) - last_index) + len(row_close_tag), 1)
+                args["end"] = f.tell()
+                if break_condition:
```

Comment:
break_condition is used only here, for readability may be better to put it here?